python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -0,0 +1,450 @@
+from collections import defaultdict
+from typing import (
+    TYPE_CHECKING,
+    AbstractSet,
+    Dict,
+    FrozenSet,
+    Iterable,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+)
+
+import pendulum
+
+import dagster._check as check
+from dagster._core.definitions.auto_materialize_policy import AutoMaterializePolicy
+from dagster._core.definitions.data_time import CachingDataTimeResolver
+from dagster._core.definitions.events import AssetKey, AssetKeyPartitionKey
+from dagster._core.definitions.freshness_based_auto_materialize import (
+    determine_asset_partitions_to_auto_materialize_for_freshness,
+)
+from dagster._core.definitions.run_request import RunRequest
+from dagster._core.definitions.time_window_partitions import (
+    get_time_partitions_def,
+)
+from dagster._utils.cached_method import cached_method
+
+from .asset_daemon_cursor import AssetDaemonCursor
+from .asset_graph import AssetGraph
+from .auto_materialize_condition import (
+    AutoMaterializeAssetEvaluation,
+    AutoMaterializeCondition,
+    AutoMaterializeDecisionType,
+    MaxMaterializationsExceededAutoMaterializeCondition,
+    MissingAutoMaterializeCondition,
+    ParentMaterializedAutoMaterializeCondition,
+    ParentOutdatedAutoMaterializeCondition,
+)
+from .partition import (
+    PartitionsDefinition,
+    ScheduleType,
+)
+
+if TYPE_CHECKING:
+    from dagster._core.instance import DagsterInstance
+    from dagster._utils.caching_instance_queryer import CachingInstanceQueryer  # expensive import
+
+
+def _will_materialize_for_conditions(
+    conditions: Optional[AbstractSet[AutoMaterializeCondition]],
+) -> bool:
+    """Based on a set of conditions, determine if the asset will be materialized."""
+    return (
+        AutoMaterializeDecisionType.from_conditions(conditions)
+        == AutoMaterializeDecisionType.MATERIALIZE
+    )
+
+
+class AssetDaemonContext:
+    def __init__(
+        self,
+        instance: "DagsterInstance",
+        asset_graph: AssetGraph,
+        stored_cursor: AssetDaemonCursor,
+        target_asset_keys: Optional[AbstractSet[AssetKey]] = None,
```

Comment:
I can do the diff to remove this and migrate the tests

Reply:
that would be awesome, ty 🙏